pkg/awsclient/client.go

```diff
@@ -376,12 +378,21 @@ func NewClient(awsAccessID, awsAccessSecret, token, region string) (Client, erro
 	}, nil
 }
 
-// GetAWSClient generates an awsclient
+// Builder implementations know how to produce a Client.
+type Builder interface {
+	GetClient(kubeClient kubeclientpkg.Client, input NewAwsClientInput) (Client, error)
+}
+
+// RealBuilder is a Builder implementation that knows how to produce a real AWS Client (i.e. one
+// that really talks to the AWS APIs).
+type RealBuilder struct{}
```

Comment:
The only comment I have on this is the RealBuilder struct. I am not in love with the naming convention here. Not sure if there is anything a better we could use.

Reply:
Outside of this implementation looks solid and hopefully will make testing easier on these reconciles.

Reply:
> The only comment I have on this is the RealBuilder struct. I am not in love with the naming convention here. Not sure if there is anything a better we could use.

Yeah, I agree it's not ideal. I played with a couple of other ideas (e.g. calling the interface `IBuilder` so I could call the real builder `Builder`) but ended up hating this one the least. Happy to take suggestions.

Reply:
I do like `IBuilder `, `Builder `better maybe even `BuilderInt `and `Builder`

Reply:
`BuilderInt` is confusing IMO, could be associated with integer. I vote for the `IBuilder` for the interface and Builder for the real obj

Reply:
Roger(io) that.

I'm going to do it in a followon PR, since this one is already reviewed for content, and to avoid having to un-conflict the two PRs that are riding on top of this one. Cool?

Reply:
awesome @2uasimojo, thanks